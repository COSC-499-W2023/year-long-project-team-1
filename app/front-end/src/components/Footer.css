/*
 * Created on Sun Oct 29 2023
 * Author: Connor Doman
 */

.site-footer {
    background-color: var(--pf-v5-global--primary-color--100);
    text-align: center;
    min-height: 20vh;
    overflow: hidden;
    display: flex;
    justify-content: space-around;
    align-items: center;
}

.site-footer .footer-item {
    text-align: center;
    color: white;
    width: 30%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    font-weight: bold;
    min-height: 3.25rem;
}

.site-footer .footer-logo {
    width: 4rem;
    height: auto;
}

.site-footer .footer-item .contact {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    gap: 1rem;
}

.site-footer .contact img {
    max-width: 2rem;
    max-height: 2rem;
}

.site-footer .footer-item .footer-links {
    display: flex;
    align-items: center;
    gap: 0.8rem;
}

.site-footer .footer-item .footer-links > a {
    font-weight: 300;
    white-space: nowrap;
}

.site-footer .footer-item .footer-links > a:link {
    color: white;
}

.site-footer .footer-item .footer-links > a:visited {
    color: white;
}

.site-footer .footer-item .footer-links > a:hover {
    color: #f58658;
}
