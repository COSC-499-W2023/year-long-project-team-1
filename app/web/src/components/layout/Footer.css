/*
Copyright [2023] [Privacypal Authors]

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/


.site-footer {
    background-color: var(--pf-v5-global--primary-color--100);
    text-align: center;
    min-height: 20vh;
    overflow: hidden;
    display: flex;
    justify-content: space-around;
    align-items: center;
    bottom: 0;
    width: 100%;
    z-index: 1000;
}

.content-wrapper {
    padding-top: 60px;
    padding-bottom: 60px;
    overflow-y: auto;
    height: 100vh;
}

.content {
    padding-bottom: 100px;
}

.site-footer .footer-item {
    text-align: center;
    color: white;
    width: 30%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    font-weight: bold;
    min-height: 3.25rem;
}

.site-footer .footer-logo {
    width: 4rem;
    height: auto;
}

.site-footer .footer-item .contact {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    gap: 1rem;
}

.site-footer .contact img {
    max-width: 2rem;
    max-height: 2rem;
}

.site-footer .footer-item .footer-links {
    display: flex;
    align-items: center;
    gap: 0.8rem;
}

.site-footer .footer-item .footer-links>a {
    font-weight: 300;
    white-space: nowrap;
}

.site-footer .footer-item .footer-links>a:link {
    color: white;
}

.site-footer .footer-item .footer-links>a:visited {
    color: white;
}

.site-footer .footer-item .footer-links>a:hover {
    color: #f58658;
}
