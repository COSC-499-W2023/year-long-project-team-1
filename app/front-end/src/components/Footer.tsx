/*
 * Created on Sun Oct 29 2023
 * Author: Connor Doman
 */

import "./Footer.css";
import Link from "next/link";
import Image from "next/image";
import logo from "@assets/logo.png";
import Githublogo from "@assets/Github_logo.png";
import Youtubelogo from "@assets/Youtube_logo.png";
import Emaillogo from "@assets/Email_logo.png";

export const Footer = () => {
    return (
        <footer className="site-footer">
            <div className="footer-item">
                <Link href="/">
                    <Image alt="logo" className="footer-logo" src={logo} />
                </Link>
            </div>
            <div className="footer-item">
                Other Sites:
                <div className="footer-links">
                    <Link href="#welcomepage">Welcome</Link>
                    <Link href="#aboutus">About Us</Link>
                    <Link href="#signup">Sign Up</Link>
                    <span>|</span>
                    <Link href="/login">Log in</Link>
                </div>
            </div>
            <div className="footer-item">
                <div className="contact">
                    Follow Us
                    <Link href="https://github.com/COSC-499-W2023/year-long-project-team-1">
                        <Image alt="logo" src={Githublogo} />
                    </Link>
                    <Link href="#youtube">
                        <Image alt="logo" src={Youtubelogo} />
                    </Link>
                    <Link href="#Email">
                        <Image alt="logo" src={Emaillogo} />
                    </Link>
                </div>
            </div>
        </footer>
    );
};

export default Footer;
