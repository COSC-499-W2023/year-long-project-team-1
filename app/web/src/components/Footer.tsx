/*
 * Created on Sun Oct 29 2023
 * Author: Connor Doman
 */

import "./Footer.css";
import Link from "next/link";
import Image from "next/image";
import logo from "@assets/logo.png";
import Githublogo from "@assets/Github_logo.png";
import Youtubelogo from "@assets/Youtube_logo.png";
import Emaillogo from "@assets/Email_logo.png";

export default async function Footer() {
    return (
        <footer className="site-footer">
            <div className="footer-item">
                <Link href="/">
                    <Image alt="PrivacyPal logo" className="footer-logo" src={logo} />
                </Link>
            </div>
            <div className="footer-item">
                Other Sites:
                <div className="footer-links">
                    <Link href="#welcomepage">Welcome</Link>
                    <Link href="#aboutus">About Us</Link>
                    <span>|</span>
                    <Link href="#signup">Sign Up</Link>
                </div>
            </div>
            <div className="footer-item">
                Follow Us:
                <div className="contact">
                    <Link href="https://github.com/COSC-499-W2023/year-long-project-team-1">
                        <Image alt="GitHub logo" src={Githublogo} />
                    </Link>
                    <Link href="#youtube">
                        <Image alt="YouTube logo" src={Youtubelogo} />
                    </Link>
                    <Link href="#Email">
                        <Image alt="mail icon" src={Emaillogo} />
                    </Link>
                </div>
            </div>
        </footer>
    );
}
