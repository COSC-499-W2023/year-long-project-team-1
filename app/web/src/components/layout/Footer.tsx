/*
 * Copyright [2023] [Privacypal Authors]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import "./Footer.css";
import Link from "next/link";
import Image from "next/image";
import logo from "@assets/dark_logo_no_name.png";
import { LoginLogoutLink } from "@components/auth/link/LoginLogoutLink";
import { UserRole } from "@lib/userRole";
import PrivacyPalLogo from "./PrivacyPalLogo";
import { GithubIconImage } from "./GIthubIconImage";

interface FooterProps {
  userRole?: UserRole; // Pass the user role if available
}

export default function Footer({ userRole }: FooterProps) {
  const footerLinks = userRole ? (
    <>
      <Link href={userRole === UserRole.CLIENT ? "/user" : "/staff"} className="bold-text">
        Go to Your Hub
      </Link>
      <span>|</span>
      <LoginLogoutLink />
    </>
  ) : (
    <LoginLogoutLink />
  );

  return (
    <footer className="site-footer">
      <div className="footer-item">
        <Link href="/">
          <PrivacyPalLogo w={48} h={48} dark={true} />
        </Link>
      </div>
      <div className="footer-links">{footerLinks}</div>
      <div className="footer-item">
        Follow Us:
        <a
          href="https://github.com/COSC-499-W2023/year-long-project-team-1"
          target="_blank"
          rel="noopener noreferrer"
          className="github-icon"
        >
          <GithubIconImage />
        </a>
      </div>
    </footer>
  );
}