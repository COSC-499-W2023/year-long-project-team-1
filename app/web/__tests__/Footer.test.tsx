/*
 * Copyright [2023] [Privacypal Authors]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Footer from "@components/layout/Footer";

describe("Footer", () => {
  it("matches snapshot", () => {
    const wrapper = render(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });

  it("contains an HTML footer element", () => {
    render(<Footer />);
    expect(screen.getByRole("contentinfo")).toBeInTheDocument();
  });

  it("renders the site logo", () => {
    render(<Footer />);
    const logo = screen.getByRole("link", { name: "PrivacyPal logo" });
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute("href", "/");
  });

  it("renders the site links", () => {
    render(<Footer />);
    expect(screen.getByRole("link", { name: "Welcome" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "About Us" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "Sign Up" })).toBeInTheDocument();
  });

  it("renders social media links", () => {
    render(<Footer />);
    expect(
      screen.getByRole("link", { name: "GitHub logo" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("link", { name: "YouTube logo" }),
    ).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "mail icon" })).toBeInTheDocument();
  });
});
