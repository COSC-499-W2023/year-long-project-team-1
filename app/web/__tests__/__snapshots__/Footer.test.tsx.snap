// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="site-footer"
      >
        <div
          class="footer-item"
        >
          <a
            href="/"
          >
            <img
              alt="PrivacyPal logo"
              class="footer-logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </a>
        </div>
        <div
          class="footer-item"
        >
          Other Sites:
          <div
            class="footer-links"
          >
            <a
              href="#welcomepage"
            >
              Welcome
            </a>
            <a
              href="#aboutus"
            >
              About Us
            </a>
            <span>
              |
            </span>
            <a
              href="/signup"
            >
              Sign Up
            </a>
          </div>
        </div>
        <div
          class="footer-item"
        >
          Follow Us:
          <div
            class="contact"
          >
            <a
              href="https://github.com/COSC-499-W2023/year-long-project-team-1"
            >
              <img
                alt="GitHub logo"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </a>
            <a
              href="#youtube"
            >
              <img
                alt="YouTube logo"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </a>
            <a
              href="#Email"
            >
              <img
                alt="mail icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </a>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="site-footer"
    >
      <div
        class="footer-item"
      >
        <a
          href="/"
        >
          <img
            alt="PrivacyPal logo"
            class="footer-logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </div>
      <div
        class="footer-item"
      >
        Other Sites:
        <div
          class="footer-links"
        >
          <a
            href="#welcomepage"
          >
            Welcome
          </a>
          <a
            href="#aboutus"
          >
            About Us
          </a>
          <span>
            |
          </span>
          <a
            href="/signup"
          >
            Sign Up
          </a>
        </div>
      </div>
      <div
        class="footer-item"
      >
        Follow Us:
        <div
          class="contact"
        >
          <a
            href="https://github.com/COSC-499-W2023/year-long-project-team-1"
          >
            <img
              alt="GitHub logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </a>
          <a
            href="#youtube"
          >
            <img
              alt="YouTube logo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </a>
          <a
            href="#Email"
          >
            <img
              alt="mail icon"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </a>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
